src/app/components/annotations/pages/annotationSearchParameters.ts

```diff
@@ -336,6 +343,50 @@ export class AnnotationSearchParameters
     );
   }
 
+  /**
+   * Assigns a tag a numerical priority specifying how relevant the tag is to
+   * the search parameters.
+   *
+   * 5. A priority set by a user (e.g. PE wants to specify that a particular
+   *    call variant tag should be verified)
+   *
+   * 4. the tag which matches any tag used to filter the current dataset
+   *    e.g. if we're searching for "Powerful Owl" then the tag "Powerful Owl"
+   *    should be prioritized from the tag list "Powerful Owl Alarm Call,
+   *    Ninoxboobook, Powerful Owl"
+   *
+   * 3. The first common_name tag
+   *
+   * 2. The first species_name tag
+   *
+   * 1. In the worst case: the first remaining tag that does not match any of
+   *    those conditions.
+   *
+   * A higher value means a the tag is more specific to the current search.
+   *
+   * This method is a callback function so that it can encapsulate "this" as a
+   * closure when passed into services as a value.
+   */
+  public tagComparer = (tag: Tag): number => {
+    if (tag.id === this.taskTag) {
+      return 5;
+    }
+
+    if (Array.from(this.tags).includes(tag.id)) {
+      return 4;
+    }
+
+    if (tag.typeOfTag === "common_name") {
+      return 3;
+    }
+
+    if (tag.typeOfTag === "species_name") {
+      return 2;
+    }
+
+    return 1;
+  }
+
```

Comment:
I don't think this belongs here. As stated previously we have need for this function in other places. Sorting tags is not unique to search paramters

Move to annotation?

Also I think if you go with a `makeTagComparer(priorities: Tag[]): (Tag => number)` you won't need to rely on closure magic (which works but the parent object is mutable, right? so this is magic in a whole other way)